### PR TITLE
Add reusable workflow, and add secrets

### PR DESCRIPTION
**Pull request by nayres**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1202025665826447/f)

---

## Description
This ticket enables developers to publish packages to NPM with little configuration, and eliminates the need to create duplicate workflows throughout the org.

[Reusable workflow docs](https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow)

### As a developer:
**I want:**
The ability to access a reusable release workflow .yml file that allows me to publish to NPM.

**I don't want:**
To be required to create a duplicate a release workflow .yml file that exists within the GitHub org.

### Known issues:
Due to the E2E testing limitations of GitHub actions, we will have to manually test these changes.

## PR Checklist 🚀

- [X] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [X] Write a useful description (above) to give reviewers appropriate context.
- [X] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 name: Release
 
 on:
+  workflow_call:
+  secrets:
```
Comment (nayres) [inline .github/workflows/release.yml:5]: I wanted to find out a way to default these, but only inputs can be defaulted. [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputs)

Comment (BRKalow) [reply .github/workflows/release.yml:5]: Oh nice! So then we'd use [this](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idsecrets) when calling the reusable action?

Comment (nayres) [reply .github/workflows/release.yml:5]: Yes, exactly!

**Comment by BRKalow** [comment]
@nayres Can we add the same for the canary workflow as well? 😄 https://github.com/hashicorp/web-platform-packages/blob/main/.github/workflows/canary-release.yml

**Comment by nayres** [comment]
> 

Pushed!

File: .github/workflows/canary-release.yml [hunk 2]
```diff
@@ -1,6 +1,12 @@
 name: Canary Release
 
 on:
+  workflow_call:
+  secrets:
+    GITHUB_TOKEN:
```
Comment (BRKalow) [inline .github/workflows/canary-release.yml:6]: Unrelated to your change, but maybe we can align the name here and use `PUBLISH_GITHUB_TOKEN`?

Comment (nayres) [reply .github/workflows/canary-release.yml]: Agreed, change pushed.

**Review by brkalow — APPROVED** [review]
Looks great! This is going to be a really nice improvement 😄

**Pull request merged**